internal/services/automation/automation_runtime_environment_resource_test.go

```diff
@@ -0,0 +1,224 @@
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data, "Python", "3.10"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccAutomationRuntimeEnvironment_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, automation.AutomationRuntimeEnvironmentResource{}.ResourceType(), "test")
+	r := AutomationRuntimeEnvironmentResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.completePowerShell(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.completePowerShellAddPackage(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccAutomationRuntimeEnvironment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, automation.AutomationRuntimeEnvironmentResource{}.ResourceType(), "test")
+	r := AutomationRuntimeEnvironmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.completePowerShell(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (a AutomationRuntimeEnvironmentResource) requiresImport(data acceptance.TestData) string {
+	template := a.completePowerShell(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_automation_runtime_environment" "import" {
+  name                    = azurerm_automation_runtime_environment.test.name
+  resource_group_name     = azurerm_automation_runtime_environment.test.resource_group_name
+  automation_account_name = azurerm_automation_runtime_environment.test.automation_account_name
+  runtime_language        = azurerm_automation_runtime_environment.test.runtime_language
+  runtime_version         = azurerm_automation_runtime_environment.test.runtime_version
+  location                = azurerm_automation_runtime_environment.test.location
+}
+`, template)
+}
+
+func (a AutomationRuntimeEnvironmentResource) basic(data acceptance.TestData, runtime, version string) string {
+	return fmt.Sprintf(`
+
+%s
+
+resource azurerm_automation_runtime_environment "test" {
+  name                    = "acc_%[2]s_basic"
+  resource_group_name     = azurerm_resource_group.test.name
+  automation_account_name = azurerm_automation_account.test.name
+
+  runtime_language        = "%[2]s"
+  runtime_version         = "%[3]s"
+
+  location                = azurerm_resource_group.test.location
+}
+
+`, a.template(data), runtime, version)
+}
+
+func (a AutomationRuntimeEnvironmentResource) completePowerShell(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+	%s
+
+resource azurerm_automation_runtime_environment "test" {
+  name                    = "powershell_complete"
```

Comment:
Name should be prefixed with `acctest` and include some random value

```suggestion
  name                    = "acctest-powershell-complete-%[2]d"
```